pkg/occlient/occlient.go

```diff
@@ -1116,23 +1119,46 @@ func (c *Client) WaitAndGetPod(selector string) (*corev1.Pod, error) {
 		return nil, errors.Wrapf(err, "unable to watch pod")
 	}
 	defer w.Stop()
-	for {
-		val, ok := <-w.ResultChan()
-		if !ok {
-			break
-		}
-		if e, ok := val.Object.(*corev1.Pod); ok {
-			glog.V(4).Infof("Status of %s pod is %s", e.Name, e.Status.Phase)
-			switch e.Status.Phase {
-			case corev1.PodRunning:
-				glog.V(4).Infof("Pod %s is running.", e.Name)
-				return e, nil
-			case corev1.PodFailed, corev1.PodUnknown:
-				return nil, errors.Errorf("pod %s status %s", e.Name, e.Status.Phase)
+
+	podChannel := make(chan *corev1.Pod)
+	watchErrorChannel := make(chan error)
+
+	go func() {
+	loop:
+		for {
+			val, ok := <-w.ResultChan()
+			if !ok {
+				watchErrorChannel <- errors.New("watch channel was closed")
+				break loop
+			}
+			if e, ok := val.Object.(*corev1.Pod); ok {
+				glog.V(4).Infof("Status of %s pod is %s", e.Name, e.Status.Phase)
+				switch e.Status.Phase {
+				case corev1.PodRunning:
+					glog.V(4).Infof("Pod %s is running.", e.Name)
+					podChannel <- e
+					break loop
+				case corev1.PodFailed, corev1.PodUnknown:
+					watchErrorChannel <- errors.Errorf("pod %s status %s", e.Name, e.Status.Phase)
+					break loop
+				}
+			} else {
+				watchErrorChannel <- errors.New("unable to convert event object to Pod")
+				break loop
 			}
 		}
+		close(podChannel)
+		close(watchErrorChannel)
+	}()
+
+	select {
+	case val := <-podChannel:
+		return val, nil
+	case err := <-watchErrorChannel:
+		return nil, err
+	case <-time.After(waitForPodTimeOut):
+		return nil, errors.Errorf("waited %s but couldn't find pod matching '%s' sel", waitForPodTimeOut, selector)
```

Comment:
The output is a bit odd:

```sh
waited 1s but couldn't find pod matching 'deploymentconfig=wild-west-ba-wildfly-levs-odo-qnvs' sel
```

It adds "sel" at the end?

Reply:
I thought I got rid of that... Will fix now!

Reply:
Should be fixed now